File: src/python/src/two_sum.py
# https://leetcode.com/problems/two-sum/description/?envType=company&envId=facebook&favoriteSlug=facebook-thirty-days


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        num_to_idx = {}
        for i in range(len(nums)):
            current = nums[i]
            pair = target - current
            if pair in num_to_idx:
                return [num_to_idx[pair], i]
            num_to_idx[current] = i
        return []

File: src/go/problems/two_sum.go
// https://leetcode.com/problems/two-sum/
package problems

func TwoSum(nums []int, target int) []int {
	numToIdx := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if idx, exists := numToIdx[complement]; exists {
			return []int{idx, i}
		}
		numToIdx[num] = i
	}
	return []int{}
}
